app/services/IdentityService.scala

```diff
@@ -2,32 +2,29 @@ package services
 
 import com.amazonaws.regions.{Region, Regions}
 import com.gu.identity.play._
-import com.gu.monitoring.{AuthenticationMetrics, CloudWatch, RequestMetrics, StatusMetrics}
 import com.gu.memsub.{Address, NormalisedTelephoneNumber}
+import com.gu.monitoring.{AuthenticationMetrics, CloudWatch, RequestMetrics, StatusMetrics}
 import com.typesafe.scalalogging.LazyLogging
 import configuration.Config
 import model.PersonalData
+import model.error.IdentityService._
 import model.error.SubsError
-import play.api.Play.current
 import play.api.http.Status
 import play.api.libs.json._
 import play.api.libs.ws.{WSClient, WSRequest, WSResponse}
 import play.api.mvc.Cookie
-import play.api.libs.concurrent.Execution.Implicits.defaultContext
-
-import scala.language.implicitConversions
-import scala.concurrent.Future
-import scalaz.{NonEmptyList, \/}
-import model.error.IdentityService._
-import services.IdentityApiClient.{authoriseCall, identityEndpoint}
 import services.PersonalDataJsonSerialiser.convertToUser
 
-class IdentityService(identityApiClient: => IdentityApiClient) extends LazyLogging {
+import scala.concurrent.{ExecutionContext, Future}
+import scala.language.{higherKinds, implicitConversions}
+import scalaz.{Monad, NonEmptyList, \/}
+
+class IdentityService[M[_]](identityApiClient: => IdentityApiClient[M])(implicit monad: Monad[M]) extends LazyLogging {
```

Comment:
this is a M instead of a Future because the tests need to get the result straight away to check, but the real one needs a Future as it will happen later.  This was the neatest way to do it that I could think of off hand.